components/org.wso2.carbon.identity.application.authenticator.organization.login/src/main/java/org/wso2/carbon/identity/application/authenticator/organization/login/OrganizationAuthenticator.java

```diff
@@ -269,20 +267,26 @@ public AuthenticatorFlowStatus process(HttpServletRequest request, HttpServletRe
             LogoutFailedException {
 
         if (context.isLogoutRequest()) {
-            String logoutUrl = getLogoutUrl(context.getAuthenticatorProperties());
-            if (StringUtils.isNotBlank(logoutUrl)) {
-                String idTokenHint = getIdTokenHint(context);
-                String orgId;
-                if (StringUtils.isNotBlank(idTokenHint)) {
-                    orgId = getIdTokenClaim(idTokenHint, ID_TOKEN_ORG_ID_PARAM);
-                } else {
-                    orgId = getUserOrgIdFromContext(context);
-                }
-                if (StringUtils.contains(logoutUrl, ORGANIZATION_QUALIFIED_PLACEHOLDER) &&
-                        StringUtils.isNotBlank(orgId)) {
-                    logoutUrl = StringUtils.replace(logoutUrl, ORGANIZATION_ID_PLACEHOLDER, orgId);
-                    context.getAuthenticatorProperties().replace(OIDC_LOGOUT_URL, logoutUrl);
+            String idTokenHint = getIdTokenHint(context);
+            String orgId;
+            if (StringUtils.isNotBlank(idTokenHint)) {
+                orgId = getIdTokenClaim(idTokenHint, ID_TOKEN_ORG_ID_PARAM);
+            } else {
+                orgId = getUserOrgIdFromContext(context);
+            }
+            if (StringUtils.isNotBlank(orgId)) {
+                String logoutUrl;
+                try {
+                    logoutUrl = ServiceURLBuilder.create()
```

Comment:
```suggestion
                  String  logoutUrl = ServiceURLBuilder.create()
```

Before catch block we can add this line
`context.getAuthenticatorProperties().put(OIDC_LOGOUT_URL, logoutUrl);`

Reply:
Fixed